### PR TITLE
Implement remaining rest functions

### PR DESCRIPTION
**Pull request by fredpetersen**
* Update deletes, and creates new analysis
* Delete deletes
* Create creates

Just CRUD things ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by duckth** [comment]
> * Update deletes, and creates new analysis
> * Delete deletes
> * Create creates
> 
> Just CRUD things ¯_(ツ)_/¯

Here, you dropped this \

**Review by lucasfth — APPROVED** [review]
lpgtm

File: NEWgIT.Infrastructure/AnalysisRepository.cs [hunk 1]
```diff
@@ -34,43 +34,55 @@ public AnalysisRepository(GitContext context)
...
         if (analysis == null) return null!; //Null is NOT Null!!!!!
-        return new AnalysisDTO(analysis.Id, analysis.RepoIdentifier, analysis.LatestCommitHash);
+        _context.Commits.Where(commit => commit.Analysis.Id == ID).Load();
+        return new AnalysisDTO(analysis.Id, analysis.RepoIdentifier, GetCommitDTOs(analysis), analysis.LatestCommitHash);
+    }
+
+    public AnalysisDTO FindByIdentifier(string repoIdentifier)
+    {
+        var analysis = _context.Analysis.Where(analysis => analysis.RepoIdentifier == repoIdentifier).FirstOrDefault();
+        if (analysis == null) return null!; //Null is NOT Null!!!!!
```
Comment (duckth) [inline NEWgIT.Infrastructure/AnalysisRepository.cs:54]: 🗡️

Comment (duckth) [reply NEWgIT.Infrastructure/AnalysisRepository.cs:54]: delet dis

File: NEWgIT.Infrastructure/DbExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace NEWgIT.Infrastructure;
+
+public static class DbExtensions
+{
+    public static Analysis? FindByIdentifier(this DbSet<Analysis> set, string identifier) =>
+        set.FirstOrDefault(a => a.RepoIdentifier == identifier);
+}
```
Comment (duckth) [inline NEWgIT.Infrastructure/DbExtensions.cs:7]:
```suggestion
}

```

**Review by duckth — CHANGES_REQUESTED** [review]

**Review by duckth — APPROVED** [review]
<3

**Pull request merged**